ouroboros-network/src/Ouroboros/Network/Mux.hs

```diff
@@ -58,9 +58,8 @@ muxStart verMpds bearer style rescb_m = do
     tbl <- setupTbl
     tq <- atomically $ newTBQueue 100
     cnt <- newTVarM 0
-    let vfn = if style == StyleClient then muxClient
-                                      else muxServer
-    (_, udesc) <- vfn verMpds bearer
+    (_, udesc) <- if style == StyleClient then muxClient verMpds bearer
```

Comment:
why not a case expression? A note: case expressions when you don't use `_` are always safe when one is adding constructors: `ghc` will always give you a hint for a missing case (at least with `-Wall` and the always is still an exaggeration ...)

Reply:
Will address.